pkg/providers/instancetype/suite_test.go

```diff
@@ -71,31 +71,38 @@ import (
 var ctx context.Context
 var stop context.CancelFunc
 var env *coretest.Environment
-var azureEnv *test.Environment
+var azureEnv, azureEnvNonZonal *test.Environment
```

Comment:
I know we need separate vars for nonzonal vs. zonal, but it feels a bit messy to have to make two of each of these in all the test files. I don't have a good suggestion for how to fix, so I'll still approve. Just wanted to call it out in case you have some ideas for easier extensibility.

Reply:
Agreed. I did not see an obvious better (and still clean) way to implement non-zonal vs zonal, besides a separate environment. However, what I think we could (and probably should) do, is isolate the tests that need both, so that we only need to use this pattern in one place, and the rest of tests can stay unaffected.